tests/unit/trace/exporters/test_zipkin_exporter.py

```diff
@@ -202,6 +205,178 @@ def test_translate_to_zipkin_span_kind_none(self):
+        exporter_ipv4 = zipkin_exporter.ZipkinExporter(
+            service_name='my_service', ipv4=ipv4)
+        zipkin_spans_ipv4 = exporter_ipv4.translate_to_zipkin(
+            span_datas=spans_ipv4)
+
+        self.assertEqual(zipkin_spans_ipv4, expected_zipkin_spans_ipv4)
+
+        # Test ipv6 local endpoint
+        exporter_ipv6 = zipkin_exporter.ZipkinExporter(
+            service_name='my_service', ipv6=ipv6)
+        zipkin_spans_ipv6 = exporter_ipv6.translate_to_zipkin(
+            span_datas=spans_ipv6)
+
+        self.assertEqual(zipkin_spans_ipv6, expected_zipkin_spans_ipv6)
+
+    def test_translate_to_zipkin_with_annotations(self):
+        trace_id = '6e0c63257de34c92bf9efcd03927272e'
+
+        annotation_attributes = {
+            'annotation_bool': True,
+            'annotation_string': 'annotation_test',
+            'key_float': .3
+        }
+
+        import datetime
+        s = '2017-08-15T18:02:26.071158'
+        time = datetime.datetime.strptime(s, '%Y-%m-%dT%H:%M:%S.%f')
+        time_events = [
+            time_event.TimeEvent(
+                timestamp=time,
+                annotation=time_event.Annotation(
+                    description='First Annotation',
+                    attributes=annotation_attributes)),
+            time_event.TimeEvent(
+                timestamp=time,
+                message_event=time_event.MessageEvent(
+                    id='message-event-id',
+                    uncompressed_size_bytes=0,
+                )),
+        ]
+
+        spans_ipv4 = [
+            span_data_module.SpanData(
+                name='child_span',
+                context=span_context.SpanContext(trace_id=trace_id),
+                span_id='6e0c63257de34c92',
+                parent_span_id='6e0c63257de34c93',
+                attributes={'test_key': 'test_value'},
+                start_time='2017-08-15T18:02:26.071158Z',
+                end_time='2017-08-15T18:02:36.071158Z',
+                child_span_count=None,
+                stack_trace=None,
+                time_events=time_events,
+                links=None,
+                status=None,
+                same_process_as_parent_span=None,
+                span_kind=0,
+            ),
+            span_data_module.SpanData(
+                name='child_span',
+                context=span_context.SpanContext(trace_id=trace_id),
+                span_id='6e0c63257de34c92',
+                parent_span_id='6e0c63257de34c93',
+                attributes={'test_key': 1},
+                start_time='2017-08-15T18:02:26.071158Z',
+                end_time='2017-08-15T18:02:36.071158Z',
+                child_span_count=None,
+                stack_trace=None,
+                time_events=time_events,
+                links=None,
+                status=None,
+                same_process_as_parent_span=None,
+                span_kind=None,
+            ),
+        ]
+
+        trace_id = '6e0c63257de34c92bf9efcd03927272e'
```

Comment:
Why are we redefining this `trace_id`? Is this a copy and paste mistake?

Reply:
Same problem with other test cases. Corrected.